.changeset/red-hats-walk.md

```diff
@@ -0,0 +1,8 @@
+---
+'@backstage/core-components': minor
+---
+
+Checkbox tree filters are no longer available in the Table component:
+
+- Deleted the `CheckboxTree` component
+- Removed the filter type `'checkbox-tree'` from the `TableFilter` types.
```

Comment:
So `core-components` is at [stability index 1](https://backstage.io/docs/overview/stability-index#core-components-githubhttpsgithubcombackstagebackstagetreemasterpackagescore-components), which means this is fine with just a guide. But tbh I think we should be a bit nicer here and not break stuff immediately.

In this case that'd mean that we keep this PR around and set a merge date of it in two weeks or more, and then only merge and release a deprecation of this for now. The `TableFilter` would print a deprecation warning if `'checkbox-tree'` is used, and we mark the `CheckboxTree` component with `@deprecated` and print a warning there as well.

Reply:
That is a fair comment! You are right, the nice thing to do would be to deprecate them first, so I will do as you suggest. Let's say we will merge this one 30/09.